### ISSUE TITLE
ACL check fails with user containing special chars

### ISSUE THREAD

**Issue by atisne** [dokuwiki/dokuwiki#4385]
### The problem

I configured authentication with OAuth plugin (on AzureAD). The login is OK.
Default ACL for connected user is READ (`*       @user   1`).
But, when I configure an ACL (WRITE) on a page for a user, the right is not taken into account: the `edit page` tool is not available (`show pagesource` instead).
The user should be able to edit the page.

Please note, the user name returned by the auth plugin, when logged in, has a pattern like _firstname.name_my-company.com_.

### Version of DokuWiki

2024-02-14a "Kaos"

### PHP Version

8

**Comment by splitbrain** [comment]
It's not clear what's going on here. Your suggested fix for this issue addresses nameencoding in group names. Group and user names in the ACL file should be nameencoded. Comparison group and user names are name encoded in auth_aclcheck_cb().

Your fix suggests an invalid ACL file. How did you create the ACL file? Did you use the ACL manager?

Please provide:

* the raw contents of your conf/acl.auth.php file
* the username and list of group names of an affected user as reported by ?do=check

**Comment by atisne** [comment]
You're right. User and group names are encoded in the ACL file when using the plugin. So, no need to encode them when using them.
The ACL file I used provides from a migration process (LDAP to SSO). So, it needs to deal with dokuwiki name encoding to build a "valid" ACL file.
Thanks a lot for your analysis.

**Comment by atisne** [comment]
Not an issue.
